sites/svelte.dev/src/lib/server/markdown/renderer.js

```diff
@@ -0,0 +1,753 @@
+							`/// <reference types="svelte/transition" />`,
+							`/// <reference types="svelte/store" />`,
+							`/// <reference types="svelte/action" />`
+						);
+					}
+
+					if (file.includes('svelte-compiler')) {
+						injected.push('// @esModuleInterop');
+					}
+
+					if (file.includes('svelte.md')) {
+						injected.push('// @errors: 2304');
+					}
+
+					// Actions JSDoc examples are invalid. Too many errors, edge cases
+					if (file.includes('svelte-action')) {
+						injected.push('// @noErrors');
+					}
+
+					if (file.includes('typescript')) {
+						injected.push('// @errors: 2304');
+					}
+
+					// For sveltekit too
+					if (
+						source.includes('$app/') ||
+						source.includes('$service-worker') ||
+						source.includes('@sveltejs/kit/')
+					) {
+						injected.push(
+							`// @filename: ambient-kit.d.ts`,
+							`/// <reference types="@sveltejs/kit" />`
+						);
+					}
+
+					if (source.includes('$env/')) {
+						// TODO we're hardcoding static env vars that are used in code examples
+						// in the types, which isn't... totally ideal, but will do for now
+						injected.push(
+							`declare module '$env/dynamic/private' { export const env: Record<string, string> }`,
+							`declare module '$env/dynamic/public' { export const env: Record<string, string> }`,
+							`declare module '$env/static/private' { export const API_KEY: string }`,
+							`declare module '$env/static/public' { export const PUBLIC_BASE_URL: string }`
+						);
+					}
+
+					if (source.includes('./$types') && !source.includes('@filename: $types.d.ts')) {
+						// Cheap copy of the types from the sveltekit repo
+						const params = `{ slug: string; }`;
+
+						injected.push(
+							`// @filename: $types.d.ts`,
+							`import type * as Kit from '@sveltejs/kit';`,
+							`export type PageLoad = Kit.Load<{${params}}>;`,
+							`export type PageServerLoad = Kit.ServerLoad<{${params}}>;`,
+							`export type LayoutLoad = Kit.Load<{${params}}>;`,
+							`export type LayoutServerLoad = Kit.ServerLoad<{${params}}>;`,
+							`export type RequestHandler = Kit.RequestHandler<{${params}}>;`,
+							`export type Action = Kit.Action<{${params}}>;`,
+							`export type Actions = Kit.Actions<{${params}}>;`,
+							`export type EntryGenerator = () => Promise<Array<{${params}}>> | Array<{${params}}>;`
+						);
+					}
+
+					// special case — we need to make allowances for code snippets coming
+					// from e.g. ambient.d.ts
+					if (file.endsWith('30-modules.md')) {
+						injected.push('// @errors: 7006 7031');
+					}
+
+					if (file.endsWith('10-configuration.md')) {
+						injected.push('// @errors: 2307');
+					}
+
+					// another special case
+					if (source.includes('$lib/types')) {
+						injected.push(`declare module '$lib/types' { export interface User {} }`);
+					}
+
+					/**
+					 * And now for tuturials
+					 */
+					if (file.startsWith('tutorial')) {
+						injected.push('// @noErrors');
+					}
```

Comment:
Sure! It can be turned into a plugin like mechanism, like `render_markdown(file, content, { twoslash_banner: [svelte(), sveltekit()] } )`. I can keep those built into the site-kit, but in case you wish to extend it, that can be done as well

Reply:
I'm not sure we need to add it as presets to site-kit, because then we need to update site-kit again each time one of them changes - AFAICT these two banner configs are distinct, you don't need kit banners in the svelte repo and vice versa, right? The API with `twoslash_banner` makes sense to me 👍

Reply:
Fair enough! I think for that one blog post we need the Kit ones. But at the same time, its a blog post. We can get away without type checking it ¯\_(ツ)_/¯

Reply:
Done! You can pass this stuff to `twoslashBanner` argument of `render_markdown`. For now, I have these definitions just in the same file. When we move this to site-kit, these will stay here and can be passed separately then